src/test/java/uk/gov/hmcts/dm/security/domain/DomainPermissionEvaluatorTests.java

```diff
@@ -239,4 +239,38 @@ public void testRolesReadPermissionsWithClassificationRestrictedAndNotMatchingRo
             ));
     }
 
+    @Test
+    public void testDocumentRolesWithLeadingSpaceCharacter() {
+        StoredDocument storedFile = new StoredDocument();
+        storedFile.setCreatedBy("nobody");
+        storedFile.setRoles(Arrays.asList(new String[] { " valid-role" }).stream().collect(Collectors.toSet()));
```

Comment:
the user can set any value for the role, so technically yes, it can end up having a leading space. also, when there are multiple roles set in the header, the ones in the middle when split might end up with a space at the front